src/Analyzer.JsonRuleEngine.UnitTests/InequalityOperatorTests.cs

```diff
@@ -0,0 +1,160 @@
+        [DataRow(2, 1, true, false, false, DisplayName = "An integer is not less than another integer")]
+        [DataRow(1, 1, true, false, false, DisplayName = "An integer is not less than another integer because both are equal")]
+        // >=int
+        [DataRow(2, 1, false, true, true, DisplayName = "An integer is greater or equal to another integer")]
+        [DataRow(1, 2, false, true, false, DisplayName = "An integer is not greater or equal to another integer")]
+        [DataRow(2, 2, false, true, true, DisplayName = "An integer is greater or equal to another integer because both are equal")]
+        // <=int
+        [DataRow(1, 2, true, true, true, DisplayName = "An integer is less or equal to another integer")]
+        [DataRow(2, 1, true, true, false, DisplayName = "An integer is not less or equal to another integer")]
+        [DataRow(1, 1, true, true, true, DisplayName = "An integer is less or equal to another integer because both are equal")]
+        // >float
+        [DataRow(2.8, 1.3, false, false, true, DisplayName = "A float is greater than another float")]
+        [DataRow(1.3, 2.8, false, false, false, DisplayName = "A float is not greater than another float")]
+        [DataRow(1.3, 1.3, false, false, false, DisplayName = "A float is not greater than another float because both are equal")]
+        // <float
+        [DataRow(1.3, 2.8, true, false, true, DisplayName = "A float is less than another float")]
+        [DataRow(2.8, 1.3, true, false, false, DisplayName = "A float is not less than another float")]
+        [DataRow(2.8, 2.8, true, false, false, DisplayName = "A float is not less than another float because both are equal")]
+        // >=float
+        [DataRow(2.8, 1.3, false, true, true, DisplayName = "A float is greater or equal to another float")]
+        [DataRow(1.3, 2.8, false, true, false, DisplayName = "A float is not greater or equal to another float")]
+        [DataRow(2.8, 2.8, false, true, true, DisplayName = "A float is greater or equal to another float because both are equal")]
+        // <=float
+        [DataRow(1.3, 2.8, true, true, true, DisplayName = "A float is less or equal to another float")]
+        [DataRow(2.8, 1.3, true, true, false, DisplayName = "A float is not less or equal to another float")]
+        [DataRow(1.3, 1.3, true, true, true, DisplayName = "A float is less or equal to another float because both are equal")]
+        public void EvaluateExpression_ValidNumericType_ReturnsExpectedEvaluationResult(object leftValue, object rightValue, bool isNegative, bool orEquals, bool evaluationResult)
+        {
+            CompareObjects(leftValue, rightValue, isNegative, orEquals, evaluationResult);
+        }
+
+        [DataTestMethod]
+        // >date
+        [DataRow(637676928000000000, 637500672000000000, false, false, true, DisplayName = "A date is greater than another date")]
```

Comment:
Is this how the date would have to be formatted in the rule?

Reply:
No no, I wrote the tests like that because it seemed easier

`   if (token.Type == JTokenType.Date) 
            {
                value = token.Value<DateTime>().ToOADate();
             }`
             
Any date format recognized by JToken as a Date should work. That's why I added this TODO: `// TODO test and document JSON date format?`. At the time I thought we might want to discuss this but I think we really should add it to the documentation, because JSON doesn't define a date format, doing it now

Reply:
I added a comment about this in the tests too.  You can add functional parsing tests in Analyzer.JsonRuleEngine.FunctionalTests in RuleParsingTests.cs.

Reply:
Thank you! I wasn't sure where to add those tests